degica/barcelona-cli
# Vault token auto-refresh

Fixes https://github.com/degica/barcelona-cli/issues/57

Vault token expires in a certain period (1 month by default). This is a good security feature but not very useful for bcn users. To avoid sudden expiration, we can implement auto-refresh as follows.

Every time bcn command executed, the bcn cli calls Vault API to generate a new token.
This way only users who haven't used bcn for a month get expired

Implementation
* store github token to config and every time bcn runs, it call the login endpoint to update the token

## How to test
I can dm you for valut-url and endpoint
1. run ` go build && ./barcelona-cli login --auth vault --vault-token <yourgithubtoken> --vault-url <vault-url> <endpoint>
`
2. note your token in `~/.bcn/login` 
3. run `go build && ./barcelona-cli review list` in barcelona project and see token is updated
